src/ert/_c_wrappers/enkf/res_config.py

```diff
@@ -405,11 +420,88 @@ def _alloc_from_dict(self, config_dict):
 
         self.model_config = ModelConfig(
             data_root=self.config_path,
-            joblist=self.site_config.job_list,
+            joblist=self.job_list,
             refcase=self.ensemble_config.refcase,
             config_dict=config_dict,
         )
 
+    def _read_env_vars(self, user_config_dict, site_config_dict):
+        env_vars = EnvironmentVarlist()
+
+        def add_from_dict(config_dict):
+            environment_vars = config_dict.get(ConfigKeys.SETENV, [])
+
+            for item in iter(environment_vars):
+                if isinstance(item, Mapping):
+                    key = item["NAME"]
+                    value = item["VALUE"]
+                else:
+                    key, value = item
+                env_vars.setenv(key, value)
+
+            paths = config_dict.get("UPDATE_PATH", [])
+
+            for key, value in iter(paths):
+                env_vars.update_path(key, value)
+
+        add_from_dict(user_config_dict)
+        add_from_dict(site_config_dict)
+        return env_vars
+
+    def _read_install_job(self, licence_root_path, user_config_dict, site_config_dict):
+        jobs = ExtJoblist()
+
+        def add_from_dict(config_dict):
+
+            for job in config_dict.get(ConfigKeys.INSTALL_JOB, []):
+                if isinstance(job, Mapping):
+                    path = job[ConfigKeys.PATH]
+                    name = job[ConfigKeys.NAME]
+                else:
+                    name, path = job
```

Comment:
here too i would expect that we have more control over the data structure, and that this if / else should not be necessary

Reply:
One is for config dict and one is for config content

Reply:
so it is!

Reply:
same as above, leaving it at your discretion!